src/main/java/com/danielagapov/spawn/Services/ChatMessage/ChatMessageService.java

```diff
@@ -16,39 +20,54 @@
 
 @Service
 public class ChatMessageService implements IChatMessageService {
-    private final IChatMessageRepository repository;
+    private final IChatMessageRepository chatMessageRepository;
+    private final IUserRepository userRepository;
+    private final IEventRepository eventRepository;
 
     @Autowired
-    public ChatMessageService(IChatMessageRepository repository) {
-        this.repository = repository;
+    public ChatMessageService(IChatMessageRepository chatMessageRepository, IUserRepository userRepository, IEventRepository eventRepository) {
+        this.chatMessageRepository = chatMessageRepository;
+        this.userRepository = userRepository;
+        this.eventRepository = eventRepository;
     }
 
     public List<ChatMessageDTO> getAllChatMessages() {
         try {
-            return ChatMessageMapper.toDTOList(repository.findAll());
+            return ChatMessageMapper.toDTOList(chatMessageRepository.findAll());
         } catch (DataAccessException e) {
             throw new BasesNotFoundException();
         }
     }
 
     public ChatMessageDTO getChatMessageById(UUID id) {
-        return ChatMessageMapper.toDTO(repository.findById(id)
+        return ChatMessageMapper.toDTO(chatMessageRepository.findById(id)
                 .orElseThrow(() -> new BaseNotFoundException(id)));
     }
 
     public List<ChatMessageDTO> getChatMessagesByTagId(UUID tagId) {
         // TODO: change this logic later, once tags are setup.
         try {
-            return ChatMessageMapper.toDTOList(repository.findAll());
+            return ChatMessageMapper.toDTOList(chatMessageRepository.findAll());
         } catch (DataAccessException e) {
             throw new RuntimeException("Error retrieving chatMessages", e);
         }
     }
 
-    public ChatMessageDTO saveChatMessage(ChatMessageDTO chatMessage) {
+    public ChatMessageDTO saveChatMessage(ChatMessageDTO chatMessageDTO) {
         try {
-            ChatMessage chatMessageEntity = ChatMessageMapper.toEntity(chatMessage);
-            repository.save(chatMessageEntity);
+            // Fetch User and Event entities
+            User userSender = userRepository.findById(chatMessageDTO.userSenderId())
+                    .orElseThrow(() -> new BaseNotFoundException(chatMessageDTO.userSenderId()));
+            Event event = eventRepository.findById(chatMessageDTO.eventId())
+                    .orElseThrow(() -> new BaseNotFoundException(chatMessageDTO.eventId()));
+
+            // Convert DTO to Entity
+            ChatMessage chatMessageEntity = ChatMessageMapper.toEntity(chatMessageDTO, userSender, event);
```

Comment:
As stated earlier, remove function parameters, `userSender` and `event` from this `toEntity()` (and other `Mapper`) methods

Reply:
Actually, it's better to keep repository-interfacing in the service layer. What you have is good, upon re-thinking.